core/src/main/java/io/grpc/inprocess/InProcessTransport.java

```diff
@@ -550,8 +559,7 @@ public void setMessageCompression(boolean enable) {}
 
       @Override
       public void setAuthority(String string) {
-        // TODO(ejona): Do something with this? Could be useful for testing, but can we "validate"
-        // it?
+        InProcessStream.this.authority = string;
```

Comment:
Actually, it was fine as it was before (without volatile). setAuthority "May only be called before start."